scripts/performance/perfcmp

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+"""Compare 2 perf runs.
+
+To use, specify the local directories that contain
+the run information::
+
+    $ ./perfcmp /results/2016-01-01-1111/ /results/2016-01-01-2222/
+
+"""
+import os
+import json
+import argparse
+
+from colorama import Fore, Style
+from tabulate import tabulate
+
+
+class RunComparison(object):
+
+    MEMORY_FIELDS = ['average_memory', 'max_memory']
+    TIME_FIELDS = ['total_time']
+    # Fields that aren't memory or time fields, they require
+    # no special formatting.
+    OTHER_FIELDS = ['average_cpu']
+
+    def __init__(self, old_summary, new_summary):
+        self.old_summary = old_summary
+        self.new_summary = new_summary
+
+    def iter_field_names(self):
+        for field in self.TIME_FIELDS + self.MEMORY_FIELDS + self.OTHER_FIELDS:
+            yield field
+
+    def old(self, field):
+        value = self.old_summary[field]
+        return self._format(field, value)
+
+    def old_suffix(self, field):
+        value = self.old_summary[field]
+        return self._format_suffix(field, value)
+
+    def new_suffix(self, field):
+        value = self.new_summary[field]
+        return self._format_suffix(field, value)
+
+    def _format_suffix(self, field, value):
+        if field in self.TIME_FIELDS:
+            return 'sec'
+        elif field in self.OTHER_FIELDS:
+            return ''
+        else:
+            # The suffix depends on the actual value.
+            return self._human_readable_size(value)[1]
+
+    def old_stddev(self, field):
+        real_field = 'std_dev_%s' % field
+        return self.old(real_field)
+
+    def new(self, field):
+        value = self.new_summary[field]
+        return self._format(field, value)
+
+    def new_stddev(self, field):
+        real_field = 'std_dev_%s' % field
+        return self.new(real_field)
+
+    def _format(self, field, value):
+        if field.startswith('std_dev_'):
+            field = field[len('std_dev_'):]
+        if field in self.MEMORY_FIELDS:
+            return self._human_readable_size(value)[0]
+        elif field in self.TIME_FIELDS:
+            return '%-3.2f' % value
+        else:
+            return '%.2f' % value
+
+    def _human_readable_size(self, value):
```

Comment:
It is a bit unfortunate that this is repeated in multiple places now, but I am not sure if it would be good idea to be using methods cross repository scripts. So it should be fine.